catalog/app/containers/Bucket/Listing.tsx

```diff
@@ -0,0 +1,850 @@
+import cx from 'classnames'
+import * as R from 'ramda'
+import * as React from 'react'
+import AutosizeInput from 'react-input-autosize'
+import { Link } from 'react-router-dom'
+import * as M from '@material-ui/core'
+import { fade } from '@material-ui/core/styles'
+import * as DG from '@material-ui/data-grid'
+
+import { renderPageRange } from 'components/Pagination2'
+import { readableBytes } from 'utils/string'
+
+const EMPTY = <i>{'<EMPTY>'}</i>
+
+const TIP_DELAY = 1000
+
+export interface Item {
+  type: 'dir' | 'file'
+  name: string
+  to: string
+  size?: number
+  modified?: Date
+  archived?: boolean
+}
+
+function maxPartial<T extends R.Ord>(a: T | undefined, b: T | undefined) {
+  if (a) return b ? R.max(a, b) : a
+  return b
+}
+
+const computeStats = R.reduce(
+  (acc, i: Item) => ({
+    files: i.type === 'file' ? acc.files + 1 : acc.files,
+    dirs: i.type === 'dir' && i.name !== '..' ? acc.dirs + 1 : acc.dirs,
+    size: i.size ? acc.size + i.size : acc.size,
+    modified: maxPartial(i.modified, acc.modified),
+  }),
+  {
+    dirs: 0,
+    files: 0,
+    size: 0,
+    modified: undefined as Date | undefined,
+  },
+)
+
+type DataGridProps = Omit<DG.GridComponentProps, 'licenseStatus'>
+
+const DataGrid = React.memo(
+  React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(inProps, ref) {
+    const props = DG.useThemeProps({ props: inProps, name: 'MuiDataGrid' })
+    return <DG.GridComponent ref={ref} {...props} licenseStatus="Valid" />
+  }),
+)
+
+type WrappedAutosizeInputProps = { className?: string } & React.ComponentProps<
```

Comment:
You can ignore `ref` since it causes error:

```
interface WrappedAutosizeInputProps extends Omit<AutosizeInputProps, 'ref'> {
  className?: string
}
```

Reply:
thx